src/index/sparse/sparse_index_node.cc

```diff
@@ -229,8 +245,29 @@ class SparseInvertedIndexNode : public IndexNode {
     }
 
  private:
+    void
+    delete_index() {
+        if (index_ != nullptr) {
+            delete index_;
+            index_ = nullptr;
+        }
+        if (map_ != nullptr) {
+            auto res = munmap(map_, map_size_);
+            if (res != 0) {
+                throw std::runtime_error("failed to munmap when trying to delete index: " +
```

Comment:
this may cause an exception to be thrown inside `~SparseInvertedIndexNode()`

Reply:
in the current impl seems the only thing we can do is to swallow the error? we provide no method for the users to explicitly destroy all the resources used so we can only clean in the destructor